### PR TITLE
Annotate ZipUtilities with objc_direct

### PR DESCRIPTION
**Pull request by jhollida24**

### CODE REVIEW

**Comment by jhollida24** [comment]
This trims 49/342 selectors from the binary (~14.3%):

```
Before:
[tw-mbp-jholliday ios-ziputilities (master)]$ objdump -section=__objc_selrefs -macho /Users/jholliday/Library/Developer/Xcode/DerivedData/ZipUtilities-abergibyouoxxpagyapshhrfnuwl/Build/Products/Release/noz  | wc -l
     342

After:
[tw-mbp-jholliday ios-ziputilities (master)]$ objdump -section=__objc_selrefs -macho /Users/jholliday/Library/Developer/Xcode/DerivedData/ZipUtilities-abergibyouoxxpagyapshhrfnuwl/Build/Products/Release/noz  | wc -l
     293
```

File: ZipUtilities/NSStream+NOZAdditions.m
Comment (NSProgrammer) [inline ZipUtilities/NSStream+NOZAdditions.m:53]: What is this about?

**Review by NSProgrammer — CHANGES_REQUESTED** [review]
Thanks for this PR!!  Unfortunately, it cannot land as-is.  ZipUtilities requires support by older versions of Xcode (and ergo clang).  As such, you will need to conditionally annotate the objc direct attributes.  TNL and TIP do this if you need a reference.

can you make that change before I approve and merge?


Thanks Jeff!  Great to see your PR!
*reviewed with iphone while rocking 16 month old to sleep*

File: ZipUtilities/NSStream+NOZAdditions.m
Comment (jhollida24) [reply ZipUtilities/NSStream+NOZAdditions.m:53]: There are implementations of these below in the file. Nothing in the project calls those methods and they're in a section preceded with this:

`#pragma mark Undocumented CFReadStream methods`

So it seems like these are overrides of private methods from the parent class (and therefore ineligible for direct dispatch).

**Comment by jhollida24** [comment]
Added a macro that no-ops with the older tools.

Hope all is well! You gotta have the kids start helping with these CRs soon.

**Review by NSProgrammer — APPROVED** [review]

**Pull request merged**
